inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -60,71 +90,221 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const DeviceMap<Infer
+    for (auto&& workerRequest : workerRequests) {
+        workerRequest._inferRequest = { executableNetwork, executableNetwork->CreateInferRequest() };
+        auto* workerRequestPtr = &workerRequest;
+        IE_ASSERT(idleWorkerRequests.try_push(workerRequestPtr) == true);
+        workerRequest._inferRequest->SetCallback(
+            [workerRequestPtr, this, device, idleWorkerRequestsPtr] (std::exception_ptr exceptionPtr) mutable {
+                IdleGuard idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
+                workerRequestPtr->_exceptionPtr = exceptionPtr;
+                {
+                    auto capturedTask = std::move(workerRequestPtr->_task);
+                    capturedTask();
+                }
+                // try to return the request to the idle list (fails if the overall object destruction has began)
+                if (idleGuard.Release()->try_push(workerRequestPtr)) {
+                    // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
+                    // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
+                    Task t;
+                    if (_inferPipelineTasks.try_pop(t))
+                        ScheduleToWorkerInferRequest(std::move(t));
+                    else if (_inferPipelineTasksDeviceSpecific[device]->try_pop(t))
+                        ScheduleToWorkerInferRequest(std::move(t), device);
+                }
+            });
+    }
+}
+
+MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const std::string&                         modelPath,
+                                                           const InferenceEngine::CNNNetwork&         network,
+                                                           const std::vector<DeviceInformation>&      metaDevices,
+                                                           const std::string&                         strDevices,
+                                                           MultiDeviceInferencePlugin*                plugin,
+                                                           const bool                                needPerfCounters)
+                                                           : _needPerfCounters(needPerfCounters)
+                                                           , _multiPlugin(plugin)
+                                                           , _workModeIsAUTO(true) {
+    if (_multiPlugin->GetCore() == nullptr) {
+        IE_THROW() << "Please, work with MULTI device via InferencEngine::Core object";
+    }
+
+    if (modelPath.empty() && network.getFunction() == nullptr) {
+        IE_THROW() << "MULTI device supports just ngraph network representation";
+    }
+
+    _core = _multiPlugin->GetCore(); // shared_ptr that holds the Core
+    _config[MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES] = strDevices;
+    _devicePrioritiesInitial = metaDevices;
```

Comment:
if you would move this line to the init list of the costructor, can  the _devicePrioritiesInitial be made const again?